code/modules/projectiles/guns/projectile/automatic.dm

```diff
@@ -74,42 +94,102 @@
 	if(user.is_holding_offhand(src) && use_launcher)
 		launcher.unload(user)
 	else
 		..()
 
-/obj/item/gun/projectile/automatic/assault_rifle/Fire(atom/target, mob/living/user, params, pointblank=0, reflex=0)
+/obj/item/gun/projectile/automatic/assault_rifle/grenade/Fire(atom/target, mob/living/user, params, pointblank=0, reflex=0)
 	if(use_launcher)
 		launcher.Fire(target, user, params, pointblank, reflex)
 		if(!launcher.chambered)
 			switch_firemodes() //switch back automatically
 	else
 		..()
 
-/obj/item/gun/projectile/automatic/assault_rifle/update_base_icon()
-	if(ammo_magazine)
-		if(ammo_magazine.stored_ammo.len)
-			icon_state = "[get_world_inventory_state()]-loaded"
-		else
-			icon_state = "[get_world_inventory_state()]-empty"
-	else
-		icon_state = get_world_inventory_state()
-
-/obj/item/gun/projectile/automatic/assault_rifle/examine(mob/user)
+/obj/item/gun/projectile/automatic/assault_rifle/grenade/examine(mob/user)
 	. = ..()
 	if(launcher.chambered)
 		to_chat(user, "\The [launcher] has \a [launcher.chambered] loaded.")
 	else
 		to_chat(user, "\The [launcher] is empty.")
+
+/obj/item/gun/projectile/automatic/assault_rifle/grenade/toggle_safety(mob/user)
+	. = ..()
+	launcher.toggle_safety() // Do not pass user to prevent double messages.
+
+// Admin-spawn machine gun. Mech subtype is meant to be used, but this can be spawned for hulks.
+/obj/item/gun/projectile/automatic/machine
+	name = "massive machine gun"
+	desc = "The XC-67 \"Creosote\" is a massive machine gun, and ranks high on most tin-pot dictators' wish lists. Firing this thing without some sort of weapons platform is a hopeless task."
+	icon = 'icons/obj/guns/machine.dmi'
+	w_class = ITEM_SIZE_HUGE
+	force = 10
+	caliber = CALIBER_RIFLE
+	origin_tech = "{'combat':9,'materials':3}"
+	ammo_type = /obj/item/ammo_casing/rifle
+	load_method = MAGAZINE
+	magazine_type = /obj/item/ammo_magazine/rifle/drum
+	allowed_magazines = /obj/item/ammo_magazine/rifle/drum
+	accuracy = 1
+	accuracy_power = 7
+	one_hand_penalty = 3000
+	bulk = GUN_BULK_RIFLE
+
+	burst_delay = 1
+	burst = 3
+	burst_accuracy = list(0,-1,-1)
+	dispersion = list(0.0, 0.6, 1.0)
+
+	fire_delay = 0
+	autofire_enabled = 1
+
+	mag_insert_sound = 'sound/weapons/guns/interaction/batrifle_magin.ogg'
+	mag_remove_sound = 'sound/weapons/guns/interaction/batrifle_magout.ogg'
+	material = /decl/material/solid/metal/steel
+
+	var/spin_up_time = null
+	var/sound_token
+
+/obj/item/gun/projectile/automatic/machine/Fire(atom/target, mob/living/user, clickparams, pointblank, reflex, set_click_cooldown)
+	if(!spin_up_time || world.time < spin_up_time + 1 SECONDS)
+		return FALSE
+	. = ..()
+
+/obj/item/gun/projectile/automatic/machine/special_check(mob/user)
+	if(!istype(user, /mob/living))
+		return FALSE
+	if(!user.check_dexterity(DEXTERITY_WEAPONS))
+		return FALSE
+
+	var/mob/living/M = user
+	if(!istype(M, /mob/living/exosuit))
+		if(!(MUTATION_HULK in M.mutations))
```

Comment:
`&&` with previous line.

Reply:
Generally checking hulk here feels restrictive - you could use mob size instead, and remove the exosuit check entirely.